### ISSUE TITLE
[Feature] Make Career start and Career end automatic

### ISSUE THREAD

**Issue by SpedNSFW** [stashapp/stash-box#619]
**Is your feature request related to a problem? Please describe.**
A lot of the time, performers will have incorrect dates set for their career starts and ends. This can come from a site like IAFD having the date set to their last scene at all times, or from users setting it prematurely when a performer hadn't been active for a little while.
Incorrect career start and end dates are a real pain for me because I'm often using them as filters. Having to manage every performer's career dates is not feasible and I want to be able to trust that stash-box has the correct dates.

**Describe the solution you'd like**
Career start should simply be the date of their first scene. Career end should be the date of their last scene, but it should remain null until it's been at least 2 years since that scene. Having the career end always be the date of their last scene will bring us back to the original problem which is why I'm suggesting a wait time. Performers take breaks often, and when they do, they will often return within 2-3 years.

**Describe alternatives you've considered**
Not necessarily an alternative but perhaps something to add to my suggestion; keeping the fields but using them as overrides. I see this being used to override career end dates that will consider re-released scenes as the performer still being active. If a performer's last scene was in 2020 but some studio re-released another scene in 2022, that performer's career end date _should_ be 2020.

**Comment by echo6ix** [comment]
Not a bad idea to augment the user reported career start and career end values by automatically overriding them if an earlier or later scene is automatically detected.

Edit: Retracting comment after more consideration. Augmenting the career end field based on a a performer's latest scene is not necessarily accurate. The performer may in fact be inactive in accordance to the manual career end field, and their last scene on StashDB could be a re-release/redistribution with a newer date.

**Comment by sleetx** [comment]
I disagree with this proposal. It would cause wildly inaccurate career start/end dates on Stashboxes that don't carry a performer's full history of scenes. Also, industry scenes are often re-released under different studios, which would further extend an invalid career end date.

**Comment by SpedNSFW** [comment]
> I disagree with this proposal. It would cause wildly inaccurate career start/end dates on Stashboxes that don't carry a performer's full history of scenes. Also, industry scenes are often re-released under different studios, which would further extend an invalid career end date.

The last part of my suggestion with keeping the fields as manual overrides resolves both concerns.
